framework/src/org/checkerframework/framework/util/MultiGraphQualifierHierarchy.java

```diff
@@ -717,6 +717,10 @@ assert getTopAnnotation(a1) == getTopAnnotation(a2) :
         }
         return newglbs;
     }
+    
+    protected boolean annotationMirrorsAreEqual(AnnotationMirror a1, AnnotationMirror a2) {
```

Comment:
1. This is a new extension point.  It is helpful to add comments to methods like these before posting a code review so that we don't have to hunt around trying to figure out why this is needed.  (And that's going to be my first comment on the method.)
2. From what I can tell, you have added this because @GuardedBy has aliases and the GLB code doesn't take this into account.  In general, the the QualifierHierarchy never assumes that two annotations are equivalent unless they are exactly the same -- meaning that it always ignores aliases.  This seems like a bug we should systematically fix, in the future.  In the meantime, I suggest overriding the public GLB method and adding the correct implementation for the Lock Checker.  (This is the usual extension point when the standard GLB is insufficient.)